models/incremental/dim_dbt__sources.sql

```diff
@@ -28,8 +28,11 @@ fields as (
         artifact_run_id,
         artifact_generated_at,
         node_id,
+        node_database,
+        node_description,
         name,
         source_name,
+        source_loader,
```

Comment:
I would suggest that rather than adding these fields directly to the raw artifacts tables (which are going to need to remain relatively static) that we leverage the json unpacking here for these fields:

```suggestion
        node_database,
        node_json:description::string as node_description,
        name,
        source_name,
        node_json:loader::string as source_loader,
```

@NiallRees - views?

Reply:
You make a good point, though we run the risk of being inconsistent around some `node_json` fields being present in the source tables, and others being pulled out in the dbt models. It feels like we should either pull all the fields out of `node_json` in the source tables, or limit the source tables to only have the `node_json` and pull any keys out in the dbt models. The former option means frequent schema changes to the raw artifacts tables as you pointed out, so I wonder if we should go with the latter, which would mean changing the existing raw artifacts tables to contain only the `node_json`. I'm happy to take that on, what do you think @alanmcruickshank ?

Reply:
Several thoughts:
- I definitely don't think we should change the artifact tables unless we absolutely have to - if nothing else it makes upgrading much much harder. Whatever we settle on for the next release should work for a good long while.
- As for whether we should roll back some of the existing expansion done in the artifact tables I think it's a balance. There will be a performance gain when filtering on anything which comes pre-expanded, but there's a flexibility which comes from only expanding them later.
- I definitely don't think we should aspire to expand _all_ fields out because there are many and not all are necessary. Now given dbt have released a public spec, we can be relatively confident it won't change for a while - but it we're lucky, the next version might still be compatible with existing artifacts.
- I do think we should definitely expand anything out which we use as a join on filter within the dbt_artifacts package - if only for performance reasons.

I think where I settle out on this is that _consistency_ isn't the goal here, it's _balance_. Let's expand out the things that are mostly likely to be joined or filtered on for processing - and then everything else is pulled from the `node_json` as required.

Reply:
As a rough guess - I wonder if actually the only things which _need_ to be expanded out for `manifest_nodes` are probably:
- command_invocation_id
- dbt_cloud_run_id
- artifact_run_id
- artifact_generated_at
- node_id
- resource_type
- node_database
- node_schema
- name

Everything else could be unpacked in the staging models?

Reply:
Fair enough. I think for the purposes of this PR, let's expand out these additional columns in the staging models as you've suggested. On a broader note, we'll see over time how much the schema changes and if there's a need to add additional columns to the raw tables.

Reply:
> As a rough guess - I wonder if actually the only things which _need_ to be expanded out for `manifest_nodes` are probably:
> 
> * command_invocation_id
> * dbt_cloud_run_id
> * artifact_run_id
> * artifact_generated_at
> * node_id
> * resource_type
> * node_database
> * node_schema
> * name
> 
> Everything else could be unpacked in the staging models?

Seems sensible to me!

Reply:
I can do this - I'm plenty in the codebase already

Reply:
I don't have access to this branch - but I'll make a fresh version of this PR building on it. See #105